src/main/java/org/sagebionetworks/web/client/widget/discussion/modal/NewThreadModal.java

```diff
@@ -9,17 +16,33 @@
 public class NewThreadModal implements NewThreadModalView.Presenter{
 
 	private NewThreadModalView view;
+	private DiscussionForumClientAsync discussionForumClient;
+	private SynapseAlert synAlert;
+	private String forumId;
+	CallbackP<Void> newThreadCallback;
 
 	@Inject
 	public NewThreadModal(
-			NewThreadModalView view
+			NewThreadModalView view,
+			DiscussionForumClientAsync discussionForumClient,
+			SynapseAlert synAlert
 			) {
 		this.view = view;
+		this.discussionForumClient = discussionForumClient;
+		this.synAlert = synAlert;
 		view.setPresenter(this);
+		view.setAlert(synAlert.asWidget());
+	}
+
+	@Override
+	public void configure(String forumId, CallbackP<Void> newThreadCallback) {
```

Comment:
either change to Callback (without the P/parameter), or change to return the new CreateDiscussionThread.